hypha/apply/funds/templates/funds/application_base.html

```diff
@@ -85,6 +85,30 @@ <h3>{% blocktrans %}Sorry this {{ page|verbose_name }} is not accepting applicat
 {% endblock %}
 
 {% block extra_js %}
+    <script>
+        document.addEventListener("DOMContentLoaded", function () {
+            let fileInputs = document.querySelectorAll("input[type='file'][required]");
+
+            fileInputs.forEach(input => {
+                input.addEventListener("invalid", function (event) {
+
+                    event.preventDefault(); // Prevent default browser behavior
+
+                    // Find the closest dff-uploader wrapper and display an error
+                    let container = input.closest(".dff-uploader");
+                    if (container) {
+                        let errorDiv = container.querySelector(".dff-invalid-files");
+                        if (errorDiv) {
+                            errorDiv.innerText = "This field is required. Please upload a file."; // Show error message
+                            errorDiv.style.color = "red";
+                        }
+                    } else {
+                        alert("Please upload a file."); // Fallback for direct file inputs
+                    }
+                });
+            });
+        });
+    </script>
```

Comment:
Maybe we can place this code in the `file-uploads.js` file? That file is always loaded and also relates to file upload fields.

Reply:
Yes, that seems more isolated and consistent.